src/ts/localization/i18nApi.ts

```diff
@@ -0,0 +1,161 @@
+import { defaultBitmovinPlayerUiTranslations } from './i18n';
+
+interface StaticVocabulary {
+  'settings': string;
+  'settings.video.quality': string;
+  'settings.audio.quality': string;
+  'settings.audio.track': string;
+  'settings.audio.mute': string;
+  'settings.window.color': string;
+  'settings.window.opacity': string;
+  'settings.subtitles': string;
+  'settings.subtitles.font.color': string;
+  'settings.subtitles.font.opacity': string;
+  'settings.subtitles.background.color': string;
+  'settings.subtitles.background.opacity': string;
+  'settings.fullscreen': string;
+  'settings.speed': string;
+  'settings.playPause': string;
+  'settings.open': string;
+  'labels.pictureInPicture': string;
+  'labels.appleAirplay': string;
+  'labels.googleCast': string;
+  'labels.vr': string;
+  'labels.off': string;
+  'labels.auto': string;
+  'labels.back': string;
+  'labels.reset': string;
+  'labels.replay': string;
+  'labels.normal': string;
+  'labels.default': string;
+  'colors.white': string;
+  'colors.black': string;
+  'colors.red': string;
+  'colors.green': string;
+  'colors.blue': string;
+  'colors.cyan': string;
+  'colors.yellow': string;
+  'colors.magenta': string;
+  'opacity.100': string;
+  'opacity.75': string;
+  'opacity.50': string;
+  'opacity.25': string;
+  'opacity.0': string;
+  'settings.subtitles.font.size': string;
+  'settings.subtitles.font.size.50': string;
+  'settings.subtitles.font.size.75': string;
+  'settings.subtitles.font.size.100': string;
+  'settings.subtitles.font.size.150': string;
+  'settings.subtitles.font.size.200': string;
+  'settings.subtitles.font.size.300': string;
+  'settings.subtitles.font.size.400': string;
+  'settings.subtitles.characterEdge': string;
+  'settings.subtitles.characterEdge.raised': string;
+  'settings.subtitles.characterEdge.depressed': string;
+  'settings.subtitles.characterEdge.uniform': string;
+  'settings.subtitles.characterEdge.dropshadowed': string;
+  'settings.subtitles.font.family': string;
+  'settings.subtitles.font.family.monospacedserif': string;
+  'settings.subtitles.font.family.proportionalserif': string;
+  'settings.subtitles.font.family.monospacedsansserif': string;
+  'settings.subtitles.font.family.proportionalsansserif': string;
+  'settings.subtitles.font.family.casual': string;
+  'settings.subtitles.font.family.cursive': string;
+  'settings.subtitles.font.family.smallcapital': string;
+  'messages.ads.remainingTime': string;
+}
+
+export type Vocabulary<V> =  V & Partial<StaticVocabulary>;
+type StringKeyMap = {[key: string]: string};
+
+export interface BitmovinPlayerUiTranslations {
+  [key: string]: Vocabulary<StringKeyMap>;
+}
+
+export interface LocalizationConfig {
+  language: string;
+  fallbackLanguages?: string[];
```

Comment:
Why is `language` a single language, and `fallbackLanguages` an array? Why not the other way round?

I think it should be the other way round because I'd like to have a fallback that is supposedly the full set of vocabulary, e.g. english. Lets assume I have base line English `en` and partial British English `en-uk`. Then I have a default Portuguese translation `pt` and a partial Brazilian Portuguese `pt-br` where also a subset of terms is overwritten.

Now I create a configuration for British English:

```ts
{
  language: 'en-uk',
  fallbackLanguages: ['en'],
}
```

Now I want to switch the language to Brazilian: how would I switch to this?

```ts
{
  language: 'pt-br',
  fallbackLanguages: ['pt', 'en'],
}
```

Is it the responsibility of the user to create a merged vocabulary `{ ...pt, ...pt-br }`? If so, what do we have the fallback array at all?

I think this should be discussed.

Reply:
regarding this comment I agree that it is the responsibility of the user to create a merged vocabulary. Current behavior that we provide is: `en` is always the fallback because that is the only language with full static vocabulary atm. (unless user duplicates the en.json and fill everything). 
what we are doing if user defines 'en' for example we merge it with default english like `{...defaultEn, ...userEn}`.

and same goes for fallback languages, basically if user defined 'de', 'it', 'fr' on the fallback languages the merge will be `{...defaultEn, ...defaultDe, ...userFr, ...userIt, ... userDe}`. why is this important?, user might have multiple languages defined in vocabulary, fallback languages are there so that we know in which order we merge them together. if they do not exist, we merge them anyways but using `Object.keys(languages)` which then does not ensure the order of the keys is same as it is defined in the `object`...

why language is single?. because there can be only one language. (maybe I should rename it to defaultLanguage to make it more clear?). why fallBackLanguages multiple? because we give the flexibility to define the order in which fallback languages will be set(maybe I should rename it to fallbackLanguageOrder to make it more clear?)

Reply:
I completely understand what you mean. It seems to me thought that if merging is the responsibility of the user then it should be sufficient to have one single complete set of vocabulary as fallback, and the `fallbackLanguages` array seems unnecessary at all.

Reply:
:+1: you are correct. I will be removing the `fallback languages` then. 
new behavior will be: `{...defaultEn, ...userLanguage}`